### PR TITLE
added isolation and integration (GSI-532)

### PR DESCRIPTION
**Pull request by KerstenBreuer**
```
added isolation and integration of resources for every artifact of workflows.
```

### CODE REVIEW

File: src/metldata/schemapack_/transform/base.py [hunk 1]
```diff
@@ -111,7 +119,7 @@ class TransformationDefinition(Generic[Config]):
         ...,
         description=(
             "A function that checks the assumptions made about the input model."
-            " Raises a ModelAssumptionError if the assumptions are not met."
+            "Raises a ModelAssumptionError if the assumptions are not met."
```
Comment (Cito) [inline src/metldata/schemapack_/transform/base.py:122]:
```suggestion
            " Raises a ModelAssumptionError if the assumptions are not met."
```

Comment (Cito) [reply src/metldata/schemapack_/transform/base.py:122]: Is there a reason why this blank char always is removed again?

File: src/metldata/schemapack_/transform/base.py [hunk 2]
```diff
@@ -125,7 +133,7 @@ class TransformationDefinition(Generic[Config]):
         ...,
         description=(
             "A class for transforming data. Raises a DataTransformationError"
-            " if the transformation fails."
+            "if the transformation fails."
```
Comment (Cito) [inline src/metldata/schemapack_/transform/base.py:136]:
```suggestion
            " if the transformation fails."
```

File: src/metldata/schemapack_/transform/base.py [hunk 3]
```diff
@@ -260,3 +268,61 @@ def step_order(self) -> list[str]:
...
+    """A model for one isolated resource of an artifact output by a workflow
+    in the context of specific input data."""
+
+    class_name: ClassName = Field(
+        ..., description="The name of the class of the resource."
+    )
+    resource_id: ResourceId = Field(..., description="The id of the resource.")
+    datapack: DataPack = Field(
+        ...,
+        description="A rooted datapack describing the resource an all its dependencies.",
```
Comment (Cito) [inline src/metldata/schemapack_/transform/base.py:283]:
```suggestion
        description="A rooted datapack describing the resource and all its dependencies.",
```

**Review by Cito — APPROVED** [review]

File: src/metldata/schemapack_/transform/base.py (hunk 1 above)
Comment (KerstenBreuer) [reply src/metldata/schemapack_/transform/base.py]: Not sure. Maybe some formatting issue. It also is missing the plusses. Is fixed in latest commit.

**Review by Cito — APPROVED** [review]

**Pull request merged**